### PR TITLE
Make 1.0.2 release

### PR DESCRIPTION
**Pull request by kghbln**
- Internal rearrangement of code (moved class into separate file)
- Improved code formatting
- Improved file documentation

This is in preparation to switch to extension registration and refers to issue https://github.com/Hutchy68/Purge/issues/3.

### CODE REVIEW

**Comment by Hutchy68** [comment]
If this changes backwards compatibility I'd rather up the version to a major 2.xx.  Haven't tested it yet but I will try to do so tonight or sometime tomorrow.

**Comment by kghbln** [comment]
No, it does not so the version should be fine. This just a new file distribution. I found out that we can do a version which would be able to serve both old and new system. That should be a 2.x.x and the one breaking with the old system should be 3.x.x. The latter fits best with with MW 1.31+ I guess so there is still some time.

**Comment by Hutchy68** [comment]
Missed this completely. Could you test the new method Sam pr'd. I just merged it in.

File: Purge.class.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * Purge
+ *
+ * @file
+ * @ingroup Extensions
+ *
+ * @license https://www.gnu.org/copyleft/gpl.html GNU General Public License 2.0 or later
+ *
+ */
+
+class PurgeActionExtension{
```
Comment (samwilson) [inline Purge.class.php]: * Class name should match the filename.
* Space required before the brace.

File: Purge.class.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	public static function onSkinTemplateNavigation( $skin, array &$content_actions ) {
+		global $wgRequest, $wgUser;
+		// Use getRelevantTitle if present so that this will work on some special pages
+		$title = method_exists( $skin, 'getRelevantTitle' ) ?
+			$skin->getRelevantTitle() : $skin->getTitle();
+		if ( $title->getNamespace() !== NS_SPECIAL && $wgUser->isAllowed( 'purge' ) ) {
+			$action = $wgRequest->getText( 'action' );
+			$content_actions['actions']['purge'] = array(
```
Comment (samwilson) [inline Purge.class.php]: Short array syntax should be used.

**Comment by kghbln** [comment]
1.0.2 is already out, thus abandoning.

**Pull request closed without merging**